service/src/main/java/uk/gov/hmcts/reform/fpl/service/document/ManageDocumentService.java

```diff
@@ -408,6 +429,75 @@ public UUID getSelectedRespondentId(CaseData caseData) {
         return getDynamicListSelectedValue(caseData.getRespondentStatementList(), mapper);
     }
 
+    public <T extends ConfidentialBundle> List<Element<DocumentWithConfidentialAddress>>
+        getDocWithConfidentialAddrFromConfidentialBundleElements(
+                CaseData caseData,
+                List<Element<T>> updatedDocuments) {
+        return getDocWithConfidentialAddrFromConfidentialBundles(caseData, updatedDocuments.stream()
+            .map(Element::getValue)
+            .collect(Collectors.toList()));
+    }
+
+    public <T extends ConfidentialBundle> List<Element<DocumentWithConfidentialAddress>>
```

Comment:
Wonder if we can simplify some of this. Could we not delegate the ability to return the confidential documents to a specific class e.g. maybe the HearingCourtBundle class which would return the documents belonging to it which are flagged as having confidential addresses?

Reply:
Not sure about this one. 
We need to display the list of uploaded documents with confidential address.
HearingCourtbundle, RespondentStatement, AdditionalApplicationsBundle, etc... all of these bundles has different data structure.